Automattic/wp-calypso
# Add/plans info popover tracking

#### Changes proposed in this Pull Request

* Add `onOpen` and `onClose` to the `InfoPopover` component as event properties.
* Convert feature slugs from symbols to strings, to (1) conform to the interface for ``, and (2) to make them easier to serialize into Tracks events.
* Add the `calypso_plans_infopopover_open` Tracks event to fire when info popovers are opened on the Plans page, with parameters for the current site ID and the specific item whose popover was opened.

Fixes `1196341175636977-as-1197867354211914`.

#### Testing instructions

Test the following on **Jetpack Cloud**, **Calypso**, and **Jetpack Connect**, enabling the corresponding variation of the `jetpackConversionRateOptimization` A/B test.

A Tracks event should fire for each of these events, with the following parameters:
* `name`: `calypso_plans_infopopover_open`
* `site_id`: the current site ID
* `item_slug`: a slug that corresponds to the info popover you clicked

**TIP:** You can enable debugging to examine Tracks events by running `localStorage.debug = 'calypso:analytics'` in your browser's console.

##### v0 / v1

* Clicking popovers next to feature items under selected plans (see: Jetpack Security, Jetpack Complete)
* Clicking the popover next to the record count for Jetpack Search (`item_slug`: `jetpack-search-record-count`)

<img width="350" alt="image" src="https://user-images.githubusercontent.com/670067/97738151-f493f880-1aab-11eb-8cca-8d336886f44e.png"> <img width="350" alt="image" src="https://user-images.githubusercontent.com/670067/97738049-d6c69380-1aab-11eb-9b38-77c463c49eba.png">

##### v2

* Clicking popovers next to feature items under selected plans (see: Jetpack Security Daily/Real-time, Jetpack Complete)
* Clicking the popover next to the price listing for Jetpack Search (`item_slug`: `jetpack-search-record-count`)

<img width="350" alt="image" src="https://user-images.githubusercontent.com/670067/97739185-7afd0a00-1aad-11eb-8378-cf217a5a7d3c.png"> <img width="350" alt="image" src="https://user-images.githubusercontent.com/670067/97739706-33c34900-1aae-11eb-8ad2-85b0b221b5a9.png">
